### PR TITLE
Remove unused MIOBuffer constructor, add explicit to another.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

File: include/iocore/eventsystem/IOBuffer.h [hunk 1]
```diff
@@ -1236,9 +1236,7 @@ class MIOBuffer
 
   const char *_location = nullptr;
 
-  MIOBuffer(void *b, int64_t bufsize, int64_t aWater_mark);
-  // cppcheck-suppress noExplicitConstructor; allow implicit conversion
-  MIOBuffer(int64_t default_size_index);
```
Comment (JosiahWI) [inline include/iocore/eventsystem/IOBuffer.h:1241]: What's the reason for making this explicit now?

Comment (JosiahWI) [reply include/iocore/eventsystem/IOBuffer.h:1241]: Makes sense. It's strange someone suppressed the linter check before instead of fixing it.

**Review by JosiahWI — COMMENTED** [review]
Looks good to me. Needs a non-Yahoo approval.

File: include/iocore/eventsystem/IOBuffer.h (hunk 1 above)
Comment (ywkaras) [reply include/iocore/eventsystem/IOBuffer.h:1241]: I would guess that code was written before the `explicit` keyword was available in whatever compiler they were using to build ATS at the time.

**Review by cmcfarlen — APPROVED** [review]
Nice cleanup

**Pull request merged**